scripts/developer_tools/n_check_github_actions.py

```
@@ -0,0 +1,159 @@
+    return "\n".join(lines)
+
+
+def watch_run(run_id: str) -> int:
+    """Watch a run until it completes, returning its exit status."""
+    result = subprocess.run(["gh", "run", "watch", run_id, "--exit-status"], check=False)
+    return result.returncode
+
+
+def view_failed_log(run_id: str) -> None:
+    """Print the failed-step logs for a run."""
+    subprocess.run(["gh", "run", "view", run_id, "--log-failed"], check=False)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)